### ISSUE TITLE
Do not report usage of deprecated parameters inside declaring method

### ISSUE THREAD

**Issue by ST-DDT** [gund/eslint-plugin-deprecation#74]
The following code should not report/require suppression comments for deprecation errors:

````ts
function demo(
  options: {
    /**
     * @deprecated
     */
    deprecated?: boolean;
    replacement?: boolean;
  } = {}
): boolean {
  const {
    // eslint-disable-next-line deprecation/deprecation
    deprecated = false,
    // eslint-disable-next-line deprecation/deprecation
    replacement = deprecated,
  } = options;
  return replacement;
}
````

**Comment by gund** [comment]
Why do you think it should not report usage of the deprecated code in your example if this plugin's job is to deprecate any usage of a deprecated code?
When you destructure a property from an object you essentially accessing/using it and so if it deprecated it must be reported.
What is your reasoning here?

**Comment by ST-DDT** [comment]
Because the method itself declares it as deprecated.
So just by adding the deprecated jsdoc tag to the method option you are required to disable the rule in that method (partially).
(if it was possible to remove the option, then the user would have done so in the first place)

And even if that is worth arguing about, the second required disable comment, is wrong because it doesn't refer to the `deprecated` option but the `deprecated` local variable.

See VSCode's code-highlighting:

![grafik](https://github.com/gund/eslint-plugin-deprecation/assets/1579362/15ed187a-ca00-4120-9314-29a3f67933be)

It doesn't mark/strike-through the value as deprecated.

**Comment by ST-DDT** [comment]
I created a separate issue, in case anybody has the same opinion/suggestion as I.

- https://github.com/gund/eslint-plugin-deprecation/issues/75
